File: 402-RemoveKDigits/402-RemoveKDigits.py
# Last updated: 07/04/2026, 11:39:10

class Solution:
    def removeKdigits(self, num: str, k: int) -> str:
        if k == len(num):
            return '0'
        stack = []
        ans = ''
        for i in num:
            while stack and k >0 and int(stack[-1]) > int(i):
                stack.pop()
                k -=1 
            stack.append(i)

        while k >0 :
            stack.pop()
            k-=1
            


        

        ans = ''.join(stack).lstrip('0')
        if len(ans):
            return ans 
        return '0'

File: 402-RemoveKDigits/402-RemoveKDigits.js
// Last updated: 07/04/2026, 11:39:05
/**
 * @param {string} num
 * @param {number} k
 * @return {string}
 */
var removeKdigits = function(num, k) {
    
     if (k === num.length) {
            return "0";
        }
        
        let stack = [];
        
        for (let digit of num) {
            while (stack.length > 0 && k > 0 && stack[stack.length - 1] > digit) {
                stack.pop();
                k--;
            }
            stack.push(digit);
        }
        
        while (k > 0) {
            stack.pop();
            k--;
        }
        
        let ans = stack.join('').replace(/^0+/, '');
        
        return ans.length ? ans : "0";
    
};